File: LLM-Translations/GPT-4/avatar/Java/Go/atcoder_ARC086_B.go

package main

import (
	"fmt"
	"bufio"
	"os"
	"strconv"
	"strings"
)

func main() {
	scanner := bufio.NewScanner(os.Stdin)
	scanner.Scan()
	n, _ := strconv.Atoi(scanner.Text())
	scanner.Scan()
	a := make([]int, n)
	for i, v := range strings.Split(scanner.Text(), " ") {
		a[i], _ = strconv.Atoi(v)
	}
	r := make([]string, 0)
	m := 0
	for i := 1; i < n; i++ {
		if abs(a[m]) < abs(a[i]) {
			m = i
		}
	}
	for i := 0; i < n; i++ {
		if (a[m] >= 0) != (a[i] >= 0) {
			r = append(r, fmt.Sprintf("%d %d", m+1, i+1))
		}
	}
	if a[m] >= 0 {
		for i := 1; i < n; i++ {
			r = append(r, fmt.Sprintf("%d %d", i, i+1))
		}
	} else {
		for i := n; i > 1; i-- {
			r = append(r, fmt.Sprintf("%d %d", i, i-1))
		}
	}
	fmt.Println(len(r))
	for _, v := range r {
		fmt.Println(v)
	}
}

func abs(x int) int {
	if x < 0 {
		return -x
	}
	return x
}
// //End of Code

This Go code reads input from the console, processes the data, and outputs the result to the console, similar to the Java code. The key difference is that Go uses a scanner to read input, and strconv package to convert strings to integers, while Java uses a Scanner object to directly read and convert input.

File: LLM-Translations/GPT-4/avatar/Java/Python/atcoder_ARC086_B.py
import sys
from typing import List, Tuple

def solve(n: int, a: List[int]) -> Tuple[int, List[Tuple[int, int]]]:
    r = []
    m = 0 
    for i in range(1, n):
        if abs(a[m]) < abs(a[i]):
            m = i 
    for i in range(0, n): 
        if a[m] >= 0 ^ a[i] >= 0:
            r.append((m + 1 , i + 1))
    if a[m] >= 0:
        for i in range(1, n):
            r.append((i, i + 1))
    else:
        for i in range(n, 1, -1):
            r.append((i , i - 1))
    return len(r), r

def main():
    n = int(sys.stdin.readline())
    a = list(map(int, sys.stdin.readline().split()))
    l, r = solve(n, a)
    print(l)
    for x in r:
        print(*x)

if __name__ == "__main__":
    main()


